libats/src/main/scala/com/advancedtelematic/libats/http/ErrorHandler.scala

```diff
@@ -68,18 +72,40 @@ object Errors {
     ).foldLeft(PartialFunction.empty[Throwable, Route])(_ orElse _)
 }
 
+trait DeployEnvSupport {
+  protected lazy val isProd =
+    Option(System.getenv("DEPLOY_ENV")).exists(env => env == "production" || env.isEmpty)
+}
 
-object ErrorHandler {
+object ErrorHandler extends DeployEnvSupport {
   import Directives._
-  import Json.obj
+
+  def logError(log: LoggingAdapter, error: Throwable): UUID = {
+    val id = UUID.randomUUID()
+    log.error(error, s"Request error $id")
+    id
+  }
+
+  def errorRepr(id: UUID, error: Throwable): Json = {
+    if(isProd)
+      Json.obj(
+        "error_id" -> id.asJson,
+        "description" -> "an error occurred".asJson
+      )
+    else
+      Json.obj(
+        "error_id" -> id.asJson,
+        "description" ->  Json.fromString(Option(error.getMessage).getOrElse("<empty error description>"))
+      )
+  }
 
   private def defaultHandler(): ExceptionHandler =
     Errors.handleAllErrors orElse ExceptionHandler {
       case e: Throwable =>
         (extractLog & extractUri) { (log, uri) =>
-          log.error(e, s"Request to $uri error")
-          val entity = obj("error" -> Json.fromString(Option(e.getMessage).getOrElse("")))
-          complete(HttpResponse(InternalServerError, entity = entity.toString()))
+          val errorId = logError(log, e)
```

Comment:
Should we also log the `uri`? Otherwise we should remove the `extractUri`.

Reply:
👍